app/templates/start-confirm-address.html

```diff
@@ -1,85 +1,77 @@
-{% extends 'base-' + display_region + '.html' %}
+{%- extends 'base-' + display_region + '.html' -%}
 
-{% set messages_dict=dict(get_flashed_messages()|groupby('level')) %}
-{% set field_messages_dict=dict(messages_dict['ERROR']|groupby('field')) %}
+{%- set messages_dict=dict(get_flashed_messages()|groupby('level')) -%}
+{%- set field_messages_dict=dict(messages_dict['ERROR']|groupby('field')) -%}
 
-{% from 'components/question/_macro.njk' import onsQuestion %}
-{% from 'components/radios/_macro.njk' import onsRadios %}
-{% from 'components/button/_macro.njk' import onsButton %}
+{%- from 'components/question/_macro.njk' import onsQuestion -%}
+{%- from 'components/radios/_macro.njk' import onsRadios -%}
+{%- from 'components/button/_macro.njk' import onsButton -%}
 
-{% set form =  {
+{%- set form =  {
     'method': 'POST',
     'attributes': {
         'action': url('StartConfirmAddress:post', display_region=display_region)
     }
-} %}
+} -%}
 
-{% set form_radio_options = [
-                        {
-                            'id': 'yes',
-                            'label': {
-                                'text': _('Yes, this is the correct address')
-                            },
-                            'value': 'Yes'
-                        },
-                        {
-                            'id': 'no',
-                            'label': {
-                                'text': _('No, this is not the correct address')
-                            },
-                            'value': 'No'
-                        }
-                    ]
-%}
+{%- set form_radio_options = [
+        {
+            'id': 'yes',
+            'label': {
+                'text': _('Yes, this is the correct address')
+            },
+            'value': 'Yes'
+        },
+        {
+            'id': 'no',
+            'label': {
+                'text': _('No, this is not the correct address')
+            },
+            'value': 'No'
+        }
+    ]
+-%}
 
-{% block main %}
-    {% if messages_dict %}
-        {% include 'partials/messages.html' with context %}
-    {% endif %}
+{%- if 'no-selection' in field_messages_dict -%}
+    {%- set error_address = {'id': 'no-selection', 'text': _('Select an answer')} -%}
+{%- endif -%}
 
-    {% if display_region_warning == 'yes' %}
-        {% set region_warning_message = _('Your address is in England, so you can only complete your census in English.') %}
-        {% from "components/panel/_macro.njk" import onsPanel %}
-        {% call onsPanel({
+{%- block main -%}
+    {%- if messages_dict -%}
+        {%- include 'partials/messages.html' with context -%}
+    {%- endif -%}
+
+    {%- if display_region_warning -%}
+        {%- set region_warning_message = _('Your address is in England, so you can only complete your census in English.') -%}
```

Comment:
Is there a situation where this message would be displayed? It doesn't get displayed when entering an English UAC onto the Welsh page (because it displays in Welsh then). So is this needed?

Reply:
It is displayed if on /cy/ (welsh) version, but with a ‘E’ region UAC - it’s basically because you can’t complete EQ in welsh for an English address, so we warn them before leaving RH, as EQ will be in English - the test in line 297 of start_handlers.py will set 'display_region_warning' to True if display_region == 'cy' and region == 'E', thus displaying the message